compute_sdk/globus_compute_sdk/sdk/hardware_report.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+import logging
+import os
+import platform
+import shlex
+import shutil
+import subprocess
+import sys
+
+
+def display_name(name: str):
+    def decorator(func):
+        func.display_name = name
+        return func
+
+    return decorator
+
+
+@display_name("psutil.virtual_memory()")
+def mem_info() -> str:
+    import psutil  # import here bc psutil is installed with endpoint but not sdk
+
+    svmem = psutil.virtual_memory()
+    return "\n".join(f"{k}: {v}" for k, v in svmem._asdict().items())
+
+
+@display_name("sys.version")
+def python_version() -> str:
+    return str(sys.version)
+
+
+def _run_command(cmd: str) -> str | None:
+    logger = logging.getLogger(__name__)  # get logger at call site (ie, endpoint)
+
+    cmd_list = shlex.split(cmd)
+    arg0 = cmd_list[0]
+
+    if not shutil.which(arg0):
+        logger.info(f"{arg0} was not found in the PATH")
+        return None
+
+    try:
+        res = subprocess.run(cmd_list, timeout=30, capture_output=True, text=True)
+        if res.stdout:
+            return str(res.stdout)
+        if res.stderr:
+            return str(res.stderr)
+        logger.warning(f"Command `{cmd}` had no output on stdout or stderr")
+    except subprocess.TimeoutExpired:
+        logger.exception(f"Command `{cmd}` timed out")
+    except Exception:
+        logger.exception(f"Command `{cmd}` failed")
```

Comment:
It might be nice to return the content of these error log messages rather than `None` to give the user a reason why some info is missing. This would also enable us to add a test to check that all commands are attempted.

Reply:
Updated the returns to give users more info. Re: testing that all commands are attempted, that would also require a change to what's returned when a command isn't in the path, and reporting that in all cases would IMO be a lot of noise on systems that would never possibly have those commands (ie macOS for `lscpu`/`lshw`, AMD clusters for `nvidia-smi`)

Reply:
I'd argue that giving more clarity to the user is worth the possible noise. For example, if I ran this on a Linux worker with NVIDIA drivers installed but missing `nvidia-smi` from its path, I would want to know why I didn't get any info back. Your call.